File: app/src/main/java/com/example/retrofit_pokemon/retrofit2/Retrofit2.java
package com.example.retrofit_pokemon.retrofit2;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class Retrofit2 {

    final private PokemonAPI urls;

    public Retrofit2(){

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(Utils.URL_BASE)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        urls = retrofit.create(PokemonAPI.class);
    }

    public PokemonAPI getService(){return urls;}

}

File: app/src/main/java/com/example/retrofit_pokemon/retrofit2/Retrofit2.kt
package com.example.retrofit_pokemon.retrofit2

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class Retrofit2 {
    val service: PokemonAPI

    init {
        val retrofit = Retrofit.Builder()
                .baseUrl(Utils.URL_BASE)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        service = retrofit.create(PokemonAPI::class.java)
    }
}